web/src/main/java/org/wso2/testgrid/web/sso/TestgridSSOAgentFilter.java

```diff
@@ -0,0 +1,118 @@
+import org.wso2.carbon.identity.sso.agent.SSOAgentFilter;
+import org.wso2.carbon.identity.sso.agent.bean.SSOAgentConfig;
+
+import java.io.IOException;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+
+/**
+ * Implementation of the SSOAgent Filter which identifies the correct httpBinding
+ * to send the request to Identity Provider.
+ */
+public class TestgridSSOAgentFilter extends SSOAgentFilter {
+
+    private static Logger logger = Logger.getLogger("org.wso2.testgrid.web.sso.TestgridSSOAgentFilter");
+
+    private static final String USERNAME = "username";
+    private static final String PASSWORD = "password";
+    private static final String CHARACTER_ENCODING = "UTF-8";
+    private static Properties properties;
+    private FilterConfig filterConfig = null;
+
+    static {
+        properties = SSOContextEventListener.getProperties();
+    }
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        this.filterConfig = filterConfig;
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
+                         FilterChain filterChain) throws IOException, ServletException {
+
+        String httpBinding = servletRequest.getParameter(
+                SSOAgentConstants.SSOAgentConfig.SAML2.HTTP_BINDING);
+        if (httpBinding != null && !httpBinding.isEmpty()) {
+            if ("HTTP-POST".equals(httpBinding)) {
+                httpBinding = "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST";
+            } else if ("HTTP-Redirect".equals(httpBinding)) {
+                httpBinding = "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect";
+            } else {
+                logger.log(Level.INFO, "Unknown SAML2 HTTP Binding. Defaulting to HTTP-POST");
+                httpBinding = "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST";
+            }
+        } else {
+            logger.log(Level.INFO, "SAML2 HTTP Binding not found in request. Defaulting to HTTP-POST");
+            httpBinding = "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST";
+        }
+        SSOAgentConfig config = (SSOAgentConfig) filterConfig.getServletContext().
+                getAttribute(SSOAgentConstants.CONFIG_BEAN_NAME);
+        config.getSAML2().setHttpBinding(httpBinding);
+
+        if (StringUtils.isNotEmpty(servletRequest.getParameter(USERNAME)) &&
+                StringUtils.isNotEmpty(servletRequest.getParameter(PASSWORD))) {
+
+            String authorization = servletRequest.getParameter(USERNAME) + ":" + servletRequest.getParameter(PASSWORD);
+            // Base64 encoded username:password value
+            authorization = Base64.encode(authorization.getBytes(CHARACTER_ENCODING));
```

Comment:
Can't we use usStandardCharsets.UTF_8.name() instead of using the string constant?